pkg/controller/injector/k8s_binding_injector_test.go

```diff
@@ -17,11 +17,175 @@ limitations under the License.
+}
+
+func TestInjectTwo(t *testing.T) {
+	binding0, binding1 := getBindings()
+	cred0, cred1 := getCreds()
+
+	injector := fakeK8sBindingInjector()
+	inject(t, injector, binding0, cred0)
+	inject(t, injector, binding1, cred1)
+
+	secretsCl := injector.client.Core().Secrets(binding0.Namespace)
+	secret, err := secretsCl.Get(binding0.Name)
+	if err != nil {
+		t.Fatalf("Error when getting secret: %s", err)
+	}
+	testCredentialsInjected(t, secret.Data, cred0)
+
+	secretsCl = injector.client.Core().Secrets(binding1.Namespace)
+	secret, err = secretsCl.Get(binding1.Name)
+	if err != nil {
+		t.Fatalf("Error when getting secret: %s", err)
+	}
+	testCredentialsInjected(t, secret.Data, cred1)
+}
+
+func TestUninjectOne(t *testing.T) {
+	binding, _ := getBindings()
+	cred, _ := getCreds()
+
+	injector := fakeK8sBindingInjector()
+	inject(t, injector, binding, cred)
+	injector.Uninject(binding)
+
+	secretsCl := injector.client.Core().Secrets(binding.Namespace)
+	secret, err := secretsCl.Get(binding.Name)
+	if err == nil {
+		testCredentialsUninjected(t, secret.Data)
+	}
 }
 
-func TestUninject(t *testing.T) {
+func TestUninjectTwo(t *testing.T) {
+	binding0, binding1 := getBindings()
+	cred0, cred1 := getCreds()
+
+	injector := fakeK8sBindingInjector()
+	inject(t, injector, binding0, cred0)
+	inject(t, injector, binding1, cred1)
+
+	injector.Uninject(binding0)
+
+	// test that binding0 is gone
+	secretsCl := injector.client.Core().Secrets(binding0.Namespace)
+	secret, err := secretsCl.Get(binding0.Name)
+	if err == nil {
+		testCredentialsUninjected(t, secret.Data)
+	}
+
+	//test that binding1 is still there
+	secretsCl = injector.client.Core().Secrets(binding1.Namespace)
+	secret, err = secretsCl.Get(binding1.Name)
+	if err != nil {
+		t.Fatalf("Error when getting secret: %s", err)
+	}
+	testCredentialsInjected(t, secret.Data, cred1)
+
+	// test that binding1 is gone after uninject
+	injector.Uninject(binding1)
+
+	secretsCl = injector.client.Core().Secrets(binding1.Namespace)
+	secret, err = secretsCl.Get(binding1.Name)
+	if err == nil {
+		testCredentialsUninjected(t, secret.Data)
+	}
+}
+
+func fakeK8sBindingInjector() *k8sBindingInjector {
+	return &k8sBindingInjector{
+		client: fake.NewSimpleClientset(),
+	}
+}
+
+func getBindings() (*servicecatalog.Binding, *servicecatalog.Binding) {
+	binding0 := &servicecatalog.Binding{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:      "name0",
+			Namespace: "namespace0",
+		},
+	}
+	binding1 := &servicecatalog.Binding{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:      "name1",
+			Namespace: "namespace1",
+		},
+	}
+	return binding0, binding1
+}
+
+func getCreds() (*brokerapi.Credential, *brokerapi.Credential) {
+	cred0 := &brokerapi.Credential{
+		Hostname: "host0",
+		Port:     "123",
+		Username: "user0",
+		Password: "password!@#!@#!0)",
+	}
+	cred1 := &brokerapi.Credential{
+		Hostname: "host1",
+		Port:     "456",
+		Username: "user1",
+		Password: "password*(&*1)",
```

Comment:
I'm sorry but can you explain why you think this PR can't go in before the credential changes happen? As I understand it, the credential changes should break the current Inject code and these unit tests anyway, thus necessitating that whoever changes that will have to fix this code as well since it would cause build and test failures. That being the case, I don't really see why that stuff being finished is a prerequisite for this going in. Is there something I'm missing?

I don't really have any opinion on the Credential stuff as I haven't had the chance to look at the proposal but I don't think this is the place to discuss it. I'd recommend you talk to @bmelville or discuss it in the context of the proposal. At least I believe there's already a proposal?